### PR TITLE
Roledeps

### PR DESCRIPTION
**Pull request by cvaraldi**
##### ISSUE TYPE

 - Feature Pull Request

##### COMPONENT NAME

role/metadata

##### ANSIBLE VERSION

```
ansible-playbook 2.3.0 (roledep cae224d3d2) last updated 2017/01/22 17:31:48 (GMT +200)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```

##### SUMMARY

```
add new ROLE_SKIP_DEPENDENCIES  configuration variable (boolean, default false):
if false: act normally
if true: don't load dependencies from metadata.
this can be useful when we have two roles A and B in a playbook.
Let's say:
- hosts: localhost
  roles:
    - { role: yum, tags: ["yum","dnf"] }
    - { role: sshd, tags: ["sshd"] }
sshd depends on yum (in metadata/main.yml), but you occasionnally want to run sshd without running tasks from yum role (because it would update all packages), and you don't even want to run tasks in yum.
This of course implies tag "sshd" (in the given example) is only in sshd role.
In case of complex playbooks, this might accelerate the run drastically if one want to run a single role/tag.
```

### CODE REVIEW

File: lib/ansible/constants.py [hunk 1]
```diff
@@ -236,6 +236,7 @@ def load_config_file():
 DEFAULT_INTERNAL_POLL_INTERVAL = get_config(p, DEFAULTS, 'internal_poll_interval', None, 0.001, value_type='float')
 ERROR_ON_MISSING_HANDLER  = get_config(p, DEFAULTS, 'error_on_missing_handler', 'ANSIBLE_ERROR_ON_MISSING_HANDLER', True, value_type='boolean')
 SHOW_CUSTOM_STATS = get_config(p, DEFAULTS, 'show_custom_stats', 'ANSIBLE_SHOW_CUSTOM_STATS', False, value_type='boolean')
+DEFAULT_ROLE_SKIP_DEPENDENCIES    = get_config(p, DEFAULTS, 'role_skip_dependencies', 'ROLE_SKIP_DEPENDENCIES', False, value_type='boolean')
```
Comment (resmo) [inline lib/ansible/constants.py]: I would change the naming a bit:

~~~ python
DEFAULT_SKIP_ROLE_DEPENDENCIES    = get_config(p, DEFAULTS, 'skip_role_dependencies', 'ANSIBLE_SKIP_ROLE_DEPENDENCIES', False, value_type='boolean')

~~~

**Comment by resmo** [comment]
nice and useful addition

File: lib/ansible/constants.py (hunk 1 above)
Comment (cvaraldi) [reply lib/ansible/constants.py]: sounds good to me. Which would need to update lib/ansible/playbook/role/__init__.py too

**Comment by cvaraldi** [comment]
hmmm, I think I made a few mistakes in my git actions trying to rebase my local branch :( Still getting new to git, it's a bit confusing to me :-(

**Comment by cvaraldi** [comment]
>> will close this pull request and create a new one

**Pull request closed without merging**